src/main/java/de/tum/in/www1/artemis/repository/specs/UserSpecs.java

```diff
@@ -0,0 +1,225 @@
+package de.tum.in.www1.artemis.repository.specs;
+
+import java.util.Arrays;
+import java.util.Set;
+
+import javax.persistence.criteria.*;
+
+import org.springframework.data.jpa.domain.Specification;
+
+import de.tum.in.www1.artemis.domain.*;
+
+/**
+ * This class contains possible specifications to query for specified users.
+ */
+public class UserSpecs {
+
+    static long FILTER_EMPTY_COURSES = -1;
+
+    static String FILTER_NO_AUTHORITY = "ROLE_NO_AUTHORITY";
+
+    /**
+     * Creates the specification to match the provided search term.
+     *
+     * @param searchTerm term to match
+     * @return specification used to chain database operations
+     */
+    public static Specification<User> getSearchTermSpecification(String searchTerm) {
+        String extendedSearchTerm = "%" + searchTerm + "%";
+        return (root, query, criteriaBuilder) -> {
+            String[] columns = { User_.LOGIN, User_.EMAIL, User_.FIRST_NAME, User_.LAST_NAME };
+            Predicate[] predicates = Arrays.stream(columns).map((column) -> criteriaBuilder.like(root.get(column), extendedSearchTerm)).toArray(Predicate[]::new);
```

Comment:
```suggestion
            Predicate[] predicates = Arrays.stream(columns).map(column -> criteriaBuilder.like(root.get(column), extendedSearchTerm)).toArray(Predicate[]::new);
```
The parentheses are not needed.